default/python/universe_generation/monsters.py

```diff
@@ -115,62 +118,60 @@ def generate_monsters(monster_freq, systems):
     tracked_plan_tries = {name: 0 for name in nest_name_map.values()}
     tracked_plan_counts = {name: 0 for name in nest_name_map.values()}
     tracked_plan_valid_locations = {fp: 0 for fp in fleet_plans if fp.name() in tracked_plan_counts}
-    tracked_nest_valid_locations = {nest: 0 for nest in nest_name_map}
 
     if not fleet_plans:
         return
 
     universe = fo.get_universe()
 
+    EMPIRE_TO_MONSTER_MIN_DIST = 2
+    empire_exclusions = set(itertools.chain.from_iterable(
+        get_systems_within_jumps(e, EMPIRE_TO_MONSTER_MIN_DIST)
+        for e in home_systems))
+    systems = [s for s in _systems if s not in empire_exclusions]
```

Comment:
Hm, this discussion is moving beyond the scope of a PR comment section, so I think we better stop here. 😉 It's not an urgent issue anyway, nothing we need to resolve right now. I plan to get back to all this on the forum, once I have a bit more time (har, har... :wink:).